### ISSUE TITLE
Print the terminal window "as is"

### ISSUE THREAD

**Issue by pmor13** [mintty/mintty#1139]
Idea: make possible to print the terminal window "as is":
![image](https://user-images.githubusercontent.com/4587883/149544713-e7bd8bf3-5c5e-47f1-8e19-c787e3bc02a6.png)

**Comment by mintty** [comment]
Printing is available via control sequences, including a screen printing function.
But maybe you are more up to a screen dump feature which is also available:
In the extended context menu (Control+right-click), there are two entries that may meet your needs:
Save as Image
HTML Screen Dump

**Comment by pmor13** [comment]
> control sequences, including a screen printing function.
Where is to read on such control sequences?

> Control+right-click / Save as Image / HTML Screen Dump
Thanks!

So, is it possible to print the terminal window "as is" w/o creating an intermediate files (.png, .html, etc.)?

**Comment by mintty** [comment]
About the control sequences: https://invisible-island.net/xterm/ctlseqs/ctlseqs.html
Look for Media Copy, and you need to select a Printer in the Options menu, section Terminal.
Other export features go via a file (image or HTML), following what xterm offers.

**Comment by BrianInglis** [comment]
You could *select*/*SelectAll* and open text editor, or *Alt-PrintScreen* and open graphics editor, then paste clipboard and print, or run a print utility with the clipboard device as input e.g. `lpr < /dev/clipboard`

**Comment by mintty** [comment]
lpr isn't really usable. It needs pre-formatted input, according to the selected printer.

**Comment by pmor13** [comment]
@BrianInglis Is it possible to print the terminal window "as is" w/o an intermediate select/copy/open/paste/etc.? Example (as usual): `CTRL+P => / usual window with printers opens / => Print`.

**Comment by mintty** [comment]
Actually, HTML export has one additional feature (described in the manual):
If you hold Shift while selecting the function, the HTML will also be opened, in your browser, from where you can easily print it.
I've added the same extra for Save as Image.

**Comment by pmor13** [comment]
Thanks!

https://invisible-island.net/xterm/ctlseqs/ctlseqs.html says:
```
CSI Ps i  Media Copy (MC).
            Ps = 0  ⇒  Print screen (default).
```
So I press `ESC 0 i`, but the printer does not print the screed. Any ideas why?
The printer is selected:
![image](https://user-images.githubusercontent.com/4587883/149817777-b42d3d70-ac6e-4eb7-88ef-fdbbe017fd9d.png)

**Comment by mintty** [comment]
Those are not input sequences, they are output sequences. Also CSI is ESC [. Do this:
`echo -e "\e[0i"`

**Comment by BrianInglis** [comment]
You have to send the correct escape sequence to the terminal input and *CSI* is `\e[`,  so after enabling anything required to allow screen printing, try `printf "\e[i"` or `printf "\e[0i"`, although as previously mentioned, Windows print spooler will probably not know what to do with the ASCII stream!

**Comment by mintty** [comment]
If you don't want the printf command to appear in the printout, you can direct it to the terminal's tty from another terminal. Printing Unicode text works because mintty uses notepad as a mediator to feed it into Windows printers.

**Comment by pmor13** [comment]
@mintty Thanks! It Works as expected.

**Comment by mintty** [comment]
Released 3.5.3 (with Shift+Save-as-image function).
